### ISSUE TITLE
pulsar_broker_tests are consistently failing on an arm64 mac

### ISSUE THREAD

**Issue by imotov** [quickwit-oss/quickwit#2921]
**Describe the bug**
Every time I run the tests on my M1-base mac pulsar_broker_tests are failing with the following message:

```
failures:

---- source::pulsar_source::pulsar_broker_tests::test_doc_batching_logic stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_doc_batching_logic' panicked at 'Setup pulsar source: Connection error: fatal error when connecting to the Pulsar server

Caused by:
    fatal error when connecting to the Pulsar server', quickwit-indexing/src/source/pulsar_source.rs:834:14

---- source::pulsar_source::pulsar_broker_tests::test_multi_topic_ingestion stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_multi_topic_ingestion' panicked at 'Create source: Failed to create source `test-pulsar-source--source-RPFBm` of type `pulsar`. Cause: Connection error: fatal error when connecting to the Pulsar server

Caused by:
    fatal error when connecting to the Pulsar server

Caused by:
    0: Connection error: fatal error when connecting to the Pulsar server
    1: fatal error when connecting to the Pulsar server', quickwit-indexing/src/source/pulsar_source.rs:968:10

---- source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_multi_consumer_ingestion stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_multi_consumer_ingestion' panicked at 'Send admin request: reqwest::Error { kind: Request, url: Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(8081), path: "/admin/v2/persistent/public/default/test-pulsar-source--partitioned-multi-consumer--topic-gTKk2/partitions", query: None, fragment: None }, source: hyper::Error(IncompleteMessage) }', quickwit-indexing/src/source/pulsar_source.rs:700:14

---- source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_single_consumer_ingestion stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_single_consumer_ingestion' panicked at 'Send admin request: reqwest::Error { kind: Request, url: Url { scheme: "http", cannot_be_a_base: false, username: "", password: None, host: Some(Domain("localhost")), port: Some(8081), path: "/admin/v2/persistent/public/default/test-pulsar-source--partitioned-single-consumer--topic-HdfgV/partitions", query: None, fragment: None }, source: hyper::Error(IncompleteMessage) }', quickwit-indexing/src/source/pulsar_source.rs:700:14

---- source::pulsar_source::pulsar_broker_tests::test_topic_ingestion stdout ----
thread 'source::pulsar_source::pulsar_broker_tests::test_topic_ingestion' panicked at 'Create source: Failed to create source `test-pulsar-source--source-GQr9o` of type `pulsar`. Cause: Connection error: fatal error when connecting to the Pulsar server

Caused by:
    fatal error when connecting to the Pulsar server

Caused by:
    0: Connection error: fatal error when connecting to the Pulsar server
    1: fatal error when connecting to the Pulsar server', quickwit-indexing/src/source/pulsar_source.rs:911:10


failures:
    source::pulsar_source::pulsar_broker_tests::test_doc_batching_logic
    source::pulsar_source::pulsar_broker_tests::test_multi_topic_ingestion
    source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_multi_consumer_ingestion
    source::pulsar_source::pulsar_broker_tests::test_partitioned_topic_single_consumer_ingestion
    source::pulsar_source::pulsar_broker_tests::test_topic_ingestion

test result: FAILED. 130 passed; 5 failed; 6 ignored; 0 measured; 0 filtered out; finished in 8.70s

error: test failed, to rerun pass `-p quickwit-indexing --lib`
make[1]: *** [test-all] Error 101
make: *** [test-all] Error 2
```

**Steps to reproduce (if applicable)**

1. Run make test-all on M1 mac
2. Observe the error message above

**Expected behavior**
Tests should pass

**Configuration:**
N/A

**Comment by fmassot** [comment]
@imotov I'm on M1 and here what I observe:
- pulsar tests are passing if I start a fresh new pulsar broker instance
- pulsar tests are always failing if I stop the pulsar broker container and restart it. When I look at the pulsar broker logs, I can see that the server does not manage to start all its services. I don't know why, I can inspect that in the coming days.

**Comment by imotov** [comment]
After another series of failures, I've decided I need to do something about it to be able to run tests. So, it turned out that master of apache/pulsar already has support for arm64, but unfortunately it didn't make it to the 2.11 branch. I tried to backport it to 2.11, but it seems that something else is missing and I didn't have time to figure it out. However, I built today's master (091ee2504ffbe6ec98e354b76e7f4c045e1914aa) and tried running our tests against it and it worked perfectly.

It is quite an ugly hack since it's not 2.11, but some pre-3.0 version, but it is much better than trying to run 2.11 image on m1. 😄 So, I just wanted to share. 

If you are on arm-based Mac, it seems like a reasonable work-around, and if you don't want to spend an half an hour building it, I pushed the image to my docker hub repo:

```
docker image rm apachepulsar/pulsar:2.11.0
docker pull imotov/pulsar:temp
docker tag imotov/pulsar:temp apachepulsar/pulsar:2.11.0
```
